### ISSUE TITLE
segfault on qira docker container

### ISSUE THREAD

**Issue by charles-l** [anordal/selfdock#3]
selfdock is awesome! Works great as a lightweight sandboxing for binaries on my system, and I'll probably be using it a lot as a replacement to docker. Thanks for building a delightfully simple piece of software like this :)

I tried to really see how far I could push it when running a docker container for [qira](https://github.com/geohot/qira/tree/master/docker). 

After building the image, I ran:
```
$ docker save qira | undocker -i -o /tmp/qira
$ HOME=/root selfdock --rootfs qira run qira /bin/ls
*** program is /bin/ls with hash b79f70b18538de0199e6829e06b547e079df8842
** found 130 names
*** elf loaded
**** using /qira/tracers/qemu/qemu/x86_64-linux-user/qemu-x86_64 for 0x3e
no qira server found, starting it
*** deleting old runs
**** running /bin/ls
****** starting WEB SERVER on 0.0.0.0:3002
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
on 0 going from 1 to 86529...*** using base 0 for 0
*** WARNING, changing segment 0x400000 4315748
Failed to get [vvar] offset 0 : [Errno 2] No such file or directory: '[vvar]'
Failed to get [vdso] offset 0 : [Errno 2] No such file or directory: '[vdso]'
*** started analysis_thread 0
qira: Bus error
```

I haven't had a chance to dig into what's segfaulting yet, but I wanted to at least mark down the issue. If this is something you think should be possible to run with `selfdock` I could take a look at debugging it a bit further.

**Comment by anordal** [comment]
Thanks for the report! Something is wrong with the latest commit, apparently.

I can reproduce something simpler:

```
$ selfdock -v /tmp /tmp run true
true: Segmentation fault
```

I got it to run with the previous commit, except it uses more /tmp space than my hardcoded limit, so you also need to increase that a bit:

```diff
-    if (mount("none", "tmp", "tmpfs", MS_NOEXEC, "size=2M")) {
+    if (mount("none", "tmp", "tmpfs", MS_NOEXEC, "size=2G")) {
         perror("tmp");
```

The command I used:

`selfdock -v /tmp/qirahome /home -e HOME /home --rootfs /tmp/qira run qira /bin/ls`

**Comment by anordal** [comment]
The segfault I could reproduce was unrelated. As suspected, it was in the then-latest commit, now fixed in the next (8486d83ac999): An embarrassing copypaste in the `-v` option handling, but you weren't using it.

I can reproduce the `Bus error`, but I don't get any `qemu: uncaught target signal 11`. If I'm guessing, maybe I'm testing a newer qira image (I built mine now from qira commit 080e6b18fd).

But increasing the /tmp space is indeed what makes the mystical bus error go away in my case.

I also made it so you can get around this without patching, if you mount /tmp in any other way (such as `--tmpfs size=16M /tmp` or `-v /tmp /tmp`). The 2MiB default is quite meager.

If you agree, the fix is adding the option `--tmpfs size=16M /tmp` (you need commit 8dafd07 for that). The error isn't exactly userfriendly, but I don't think it looks like a fault of selfdock anymore.

**Comment by charles-l** [comment]
Yep, looks like using a bigger `/tmp` fixes it. And agreed, this error is the fault of the docker image, not `selfdock`. Marking closed. Thanks!
